In [1]:
import os
from os.path import join, isdir
import shutil

In [38]:
# get all folders
base_path = 'MultiPIE_sample/'
testdir = 'allimg/'
folders = [folder for folder in os.listdir(base_path) if isdir(join(base_path, folder))]

In [45]:
# copy all images in subfolders into one big _testdir_ folder
for f in folders:
    for i, img_name in enumerate(os.listdir(join(base_path, f))):
        print(join(testdir, f + '_' + str(i) + '.png'))
        shutil.copyfile(join(base_path, f, img_name), join(testdir, f + '_' + str(i) + '.png'))

allimg/001_01_041_17_0.png
allimg/001_01_041_17_1.png
allimg/001_01_190_07_0.png
allimg/001_01_190_07_1.png
allimg/001_01_080_17_0.png
allimg/001_01_080_17_1.png
allimg/001_01_050_07_0.png
allimg/001_01_050_07_1.png
allimg/001_01_080_12_0.png
allimg/001_01_080_12_1.png
allimg/001_01_190_12_0.png
allimg/001_01_190_12_1.png
allimg/001_01_051_12_0.png
allimg/001_01_051_12_1.png
allimg/001_01_140_17_0.png
allimg/001_01_140_17_1.png
allimg/001_01_130_02_0.png
allimg/001_01_130_02_1.png
allimg/001_01_080_07_0.png
allimg/001_01_080_07_1.png
allimg/001_01_130_12_0.png
allimg/001_01_130_12_1.png
allimg/001_01_190_02_0.png
allimg/001_01_190_02_1.png
allimg/001_01_050_17_0.png
allimg/001_01_050_17_1.png
allimg/001_01_050_12_0.png
allimg/001_01_050_12_1.png
allimg/001_01_190_17_0.png
allimg/001_01_190_17_1.png
allimg/001_01_051_07_0.png
allimg/001_01_051_07_1.png
allimg/001_01_140_12_0.png
allimg/001_01_140_12_1.png
allimg/001_01_130_07_0.png
allimg/001_01_130_07_1.png
allimg/001_01_041_12_0.png
a

In [58]:
# list all unique values for any of the below mentioned image properties
# 0 = person ID
# 1 = session ID
# 2 = viewpoint
# 3 = lighting
# 4 = expression
filename_part = 0
with open('all_folders.txt', 'r') as f:
    all_folders = f.read().splitlines()

viewpoints = []
for folder in all_folders:
    viewpoints += [folder.split('_')[filename_part]]

print(list(set(viewpoints)))

['164', '337', '293', '002', '022', '047', '124', '068', '123', '120', '191', '204', '084', '003', '234', '237', '064', '296', '049', '302', '310', '160', '136', '038', '344', '286', '009', '329', '335', '210', '257', '189', '309', '321', '021', '033', '223', '036', '103', '202', '318', '265', '113', '303', '016', '057', '186', '027', '072', '060', '185', '076', '224', '322', '156', '037', '065', '074', '058', '137', '004', '079', '001', '044', '077', '116', '118', '199', '163', '208', '315', '325', '248', '092', '253', '069', '029', '018', '190', '008', '343', '203', '161', '215', '075', '143', '030', '149', '059', '086', '327', '071', '214', '159', '247', '024', '145', '166', '201', '256', '032', '133', '051', '147', '097', '067', '225', '045', '188', '207', '278', '339', '140', '102', '219', '141', '228', '096', '254', '260', '174', '276', '135', '162', '231', '287', '014', '262', '263', '169', '298', '007', '211', '230', '200', '313', '153', '063', '050', '151', '091', '125', '132'

In [55]:
len(str(3))

1

In [56]:
'0'*3

'000'

# Dataset structure

Filename structure:

    <personID>_<sessionID>_<viewpoint>_<lighting>_<expression>.png
    
    PersonID:   The ID number of the person, between 001-346 (but there are be 337 in total...)
    sessionID:  Either 01, 02, 03 or 04
    viewpoint:  One out of ['051', '080', '140', '190', '041', '130', '050'], see image below for details
    lighting:   One out of ['02', '07', '12', '17']
                        '02': left side of face illuminated
                        '07': front side of face illuminated
                        '12': right side of face illuminated
                        '17': all sides of face illuminated
    expression: 0: neutral
                1: happy
                



    
![viewpoint details](report_images/camera_viewpoints.png)

# RaFD structure

In [2]:
with open('all_rafd_filenames.txt', 'r') as f:
    all_rafd = f.read().splitlines()

In [3]:
print("RaFD total images: {}".format(len(all_rafd)))

RaFD total images: 8040


In [18]:
all_viewpoints = [elem.split('_')[0][4:] for elem in all_rafd]
all_model_numbers = [elem.split('_')[1] for elem in all_rafd]
all_ethnicities = [elem.split('_')[2] for elem in all_rafd]
all_genders = [elem.split('_')[3] for elem in all_rafd]
all_emotions = [elem.split('_')[4] for elem in all_rafd]
all_gazes = [elem.split('_')[5].split('.')[0] for elem in all_rafd]

In [5]:
print("Unique viewpoints:", set(all_viewpoints))

Unique viewpoints: {'045', '180', '135', '090', '000'}


In [7]:
len(set(all_model_numbers))

67

In [8]:
max(map(int, all_model_numbers))

73

In [12]:
print("model numbers raning from 1 to {} (inclusive), with the exception of: {}".format(max(map(int, all_model_numbers)), set(range(max(map(int, all_model_numbers))+1)) - set(map(int, all_model_numbers))))

model numbers raning from 1 to 73 (inclusive), with the exception of: {0, 34, 66, 6, 13, 17, 62}


In [14]:
print("the ethicities in the data are: {}".format(set(all_ethnicities)))

the ethicities in the data are: {'Kid', 'Caucasian', 'Moroccan'}


In [15]:
print("all genders are: {}".format(set(all_genders)))

all genders are: {'female', 'male'}


In [16]:
print("all emotions are: {}".format(set(all_emotions)))

all emotions are: {'happy', 'contemptuous', 'disgusted', 'fearful', 'neutral', 'angry', 'surprised', 'sad'}


In [19]:
print("all gazes are: {}".format(set(all_gazes)))

all gazes are: {'right', 'frontal', 'left'}


# test stuff

In [59]:
from os.path import basename
from zipfile import 

In [60]:
a = 'a/b/d/c.png'
basename(a)

'c.png'